src/containers/SubjectPage/SubjectContainer.jsx

```diff
@@ -34,36 +33,104 @@ const SubjectPage = ({
   skipToContentId,
   t,
   subjectId,
+  urlTopicId,
+  urlSubTopicId,
   data,
 }) => {
-  const handleFilterClick = newValues => {
-    const searchString = `?${queryString.stringify({
-      filters: newValues.join(','),
-    })}`;
-    history.push(
-      newValues.length > 0
-        ? {
-            search: searchString,
-          }
-        : {},
-    );
-  };
+  const [topicId, setTopicId] = useState(urlTopicId);
+  const [subTopicId, setSubTopicId] = useState(urlSubTopicId);
+  const [topic, setTopic] = useState(null);
+  const [subTopic, setSubTopic] = useState(null);
+  const [currentLevel, setCurrentLevel] = useState('Subject');
 
-  const activeFilters = getFiltersFromUrlAsArray(location);
+  useEffect(() => {
+    history.replace(toTopic(subjectId, filter?.id, topicId, subTopicId));
+  }, [topic, subTopic]);
+
+  const activeFilterId = getFiltersFromUrl(location);
   const { subject = {} } = data;
   const { name: subjectName } = subject;
   const subjectpage = subject.subjectpage || {};
 
   const {
     latestContent,
-    facebook,
-    banner,
     editorsChoices,
     layout,
     about,
     metaDescription,
   } = subjectpage;
 
+  const filter = subject.filters.find(filter => filter.id === activeFilterId);
+
+  const breadCrumbs = [
+    {
+      id: subject.id,
+      label: subject.name,
+      typename: 'Subjecttype',
+      url: '#',
+    },
+    ...(filter
+      ? [
+          {
+            id: filter.id,
+            label: filter.name,
+            typename: 'Subject',
+            url: '#',
+            isCurrent: currentLevel === 'Subject',
+          },
+        ]
+      : []),
+    ...(topic ? [{ ...topic, isCurrent: currentLevel === 'Topic' }] : []),
+    ...(subTopic
+      ? [{ ...subTopic, isCurrent: currentLevel === 'Subtopic' }]
+      : []),
+  ];
+
+  const setTopicBreadCrumb = topic => {
+    setCurrentLevel('Topic');
+    setTopicId(topic.id);
+    setTopic(
+      topic
+        ? {
+            ...topic,
+            typename: 'Topic',
+            url: '#',
+          }
+        : null,
+    );
+  };
+
+  const setSubTopicBreadCrumb = topic => {
+    setCurrentLevel('Subtopic');
+    setSubTopicId(topic.id);
+    setSubTopic(
+      topic
+        ? {
+            ...topic,
+            typename: 'Subtopic',
+            url: '#',
+          }
+        : null,
+    );
+  };
+
+  const headerRef = useRef(null);
+  const mainRef = useRef(null);
+  const subRef = useRef(null);
+
+  const handleNav = (e, item) => {
+    e.preventDefault();
+    const { typename } = item;
+    setCurrentLevel(typename);
+    if (typename === 'Subjecttype') {
+      scrollToRef(headerRef);
+    } else if (typename === 'Topic') {
+      scrollToRef(mainRef);
+    } else if (typename === 'Subtopic') {
+      scrollToRef(subRef);
+    }
```

Comment:
```suggestion
   if (typename === 'Subjecttype') {
      scrollToRef(headerRef);
    } else if (typename === 'Subject') {
      scrollToRef(headerRef);
    } else if (typename === 'Topic') {
      scrollToRef(mainRef);
    } else if (typename === 'Subtopic') {
      scrollToRef(subRef);
    }
```
Må også legges inn for Subject

Reply:
I og med at Subject og Subjecttype har samme ref kunne det vore if(typename === 'Subject' || typename === 'Subjecttype') eller if(typename.includes('Subject')) .